synapse/storage/persist_events.py

```diff
@@ -460,14 +460,13 @@ async def _persist_event_batch(
                         )
 
                     for room_id, ev_ctx_rm in events_by_room.items():
-                        latest_event_ids = (
+                        latest_event_ids = set(
                             await self.main_store.get_latest_event_ids_in_room(room_id)
                         )
                         new_latest_event_ids = await self._calculate_new_extremities(
```

Comment:
Should we type `_calculate_new_extremities` to return a `set`? It seems undefined now.

Reply:
Good idea, we might as well fix that while here.